### ISSUE TITLE
Wheat farm cart pushers

### ISSUE THREAD

**Issue by cdsmith-umn** [Keriew/augustus#57]
I played Tarraco again this morning, using v1.4.0. I noticed that wheat farm cart pushers acted very strangely when they had to traverse stairs to reach granaries or warehouses. That is, they would never traverse the stairs. In addition, they would only intermittently put their goods into the warehouse that I built right next to their farms. I tried to discover a pattern to their behaviour, but I could not. The warehouse could be completely empty and they would still only put their goods into it periodically. It also didn't seem to be correlated to how much wheat I had in warehouses because the amount of wheat I had in warehouses continued to go up: 33, 48, 56, 78, 96, 116, and 126 are the numbers I saw when I checked.

Not only that, but it was **_only_** the wheat farms' cart pushers that were affected by this. I built a fruit farm next to them and that farm's cart pusher was always busy. You can see this in the screenshot I've attached. The fruit farm has a ready harvest because the cart pusher had to go a long way to get to the nearest warehouse. (It was the best I could do to optimize their paths, as greater efficiency didn't hold enough value for me.)

![wheat_farm_bug](https://user-images.githubusercontent.com/9805709/83947609-b4bb3480-a7dd-11ea-8d3b-bb8f63ac6926.png)

**Comment by Keriew** [comment]
Could you attach the save and the augustus.ini file? That would help in investigating.

**Comment by theradeonxt** [comment]
Hello, 
About the stairs thing, I remember way back before julius or augustus, original C3 I had to make a double road for stairs to get all types of walkers to pass. 
Can you try with 2 roads parallel (meaning not perpendicular like in tge picture) to the stairs and see if it improves things ?

**Comment by Keriew** [comment]
All right, I've figured out what the issue is. Will fix promptly.

**Comment by Keriew** [comment]
@cdsmith-umn Could you test the latest unstable build and check if it works?

**Comment by cdsmith-umn** [comment]
@Keriew Sorry for the lateness of the reply. I did try with the latest unstable build and the bug is definitely fixed. I hadn't saved the game in which I originally experienced that bug, but I did start Tarraco anew this morning. I put my first wheat farm over there and put the granary across the river from it, with only one roadblock between them.

I have attached a zip file with the ini and the svx file from this morning's test.

[files.zip](https://github.com/Keriew/augustus/files/4745498/files.zip)
